core/network/fabric-interop-cc/contracts/interop/membership.go

```diff
@@ -58,16 +347,157 @@ func (s *SmartContract) CreateMembership(ctx contractapi.TransactionContextInter
+	for _, attestation := range append(counterAttestedMembership.Attestations, attestedMembershipSet.Attestations...) {
+		if matchedNonce == "" {
+			matchedNonce = attestation.Nonce
+		} else {
+			if matchedNonce != attestation.Nonce {
+				return fmt.Errorf("Mismatched nonces across two attestations: %s, %s", matchedNonce, attestation.Nonce)
+			}
+		}
+	}
+
+	// Ensure valid attestations from all local IIN Agents
+	for _, attestation := range counterAttestedMembership.Attestations {
+		err = validateAttestation(attestation, counterAttestedMembership.AttestedMembershipSet + attestation.Nonce)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Validate foreign membership cert chains
+	err = validateMemberCertChains(&foreignMembership)
+	if err != nil {
+		return err
+	}
+
+	// Ensure authentic and valid attestations from all foreign IIN Agents
+	for _, attestation := range attestedMembershipSet.Attestations {
+		if attestation.UnitIdentity.SecurityDomain != foreignMembership.SecurityDomain {
+			return fmt.Errorf("Foreign agent security domain %s does not match attested membership security domain %s",
+				attestation.UnitIdentity.SecurityDomain, foreignMembership.SecurityDomain)
+		}
+		foundMemberMatch := false
+		for _, member := range foreignMembership.Members {
+			var isSignerRoot bool
+			var leafCertPEM string
+			if member.Type == "ca" {
+				leafCertPEM = member.Value
+				isSignerRoot = true
+			} else if member.Type == "certificate" {
+				leafCertPEM = member.Chain[len(member.Chain) - 1]
+				isSignerRoot = (len(member.Chain) == 1)
+			}
+			attesterCert, err := parseCert(attestation.Certificate)
+			if err != nil {
+				continue
+			}
+			leafCert, err := parseCert(leafCertPEM)
+			if err != nil {
+				continue
+			}
+			err = validateCertificateUsingCA(attesterCert, leafCert, isSignerRoot)
+			if err == nil {
+				foundMemberMatch = true
+				break
+			}
+		}
+		if !foundMemberMatch {
+			return fmt.Errorf("No matching member certificate chain found for attester")
+		}
+		// Validate signature
+		err = validateAttestation(attestation, attestedMembershipSet.Membership + attestation.Nonce)
+		if err != nil {
+			return err
+		}
+	}
+
+	membershipKey, err := ctx.GetStub().CreateCompositeKey(membershipObjectType, []string{foreignMembership.SecurityDomain})
+	_, getErr := s.GetMembershipBySecurityDomain(ctx, foreignMembership.SecurityDomain)
+	if getErr != nil {
+		return getErr
+	}
+
+	membershipBytes, err := json.Marshal(foreignMembership)
+	if err != nil {
+		return fmt.Errorf("Marshal error: %s", err)
+	}
+	return ctx.GetStub().PutState(membershipKey, membershipBytes)
+
+}
+
+// updateMembership is used by a network admin to update an existing Membership in the ledger with an unattested membership
+// TODO: Remove this function after creating Corda IIN Agents. Retaining this temporarily for backward compatibility.
+func (s *SmartContract) updateMembership(ctx contractapi.TransactionContextInterface, membershipJSON string) error {
```

Comment:
Same comment as for `createMembership` to not allow this to be directly invoked.

Reply:
Fixed.